### PR TITLE
Refactor account #103

### LINKED ISSUE #103: Refactor account

### PR DESCRIPTION
**Pull request by DmitryKus**
resolves #103

### CODE REVIEW

File: js/src/features/human-resouce/actions.ts [hunk 1]
```diff
@@ -158,8 +158,22 @@ export const closeAbsenceUpdateModal = createAction(
   "CLOSE_ABSENCE_UPDATE_MODAL",
 )();
 
-export const openAbsenceCreateForm = createAction("OPEN_ABSENCE_CREATE_FORM")();
+export const openAbsenceCreateModal = createAction(
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/actions.ts:161]: Why did you rename openAbsenceCreateForm to openAbsenceCreateModal? If you want to make the form non-modal, you will have to rename the action

File: js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationListInner.tsx [hunk 2]
```diff
@@ -1,25 +1,18 @@
 import React, { useCallback } from "react";
-import { CloseCircleOutlined, EditOutlined } from "@ant-design/icons";
+import { EditOutlined } from "@ant-design/icons";
 import { EmployeeLocation } from "LocationModels";
 
 interface Props {
   item: EmployeeLocation;
   edit: (val: EmployeeLocation) => void;
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationListInner.tsx:7]: Что значит "val" in ?

File: js/src/features/human-resouce/components/Note/NoteForm.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Form, Button } from "antd";
+import TextArea from "antd/lib/input/TextArea";
+import { FormErrorMap } from "MyTypes";
+import { useFormServerErrors } from "hooks/useFormServerErrors";
+
+interface Props {
+  addNote: (text: string) => any;
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/Note/NoteForm.tsx:8]: addNote: (text: string) => void;

File: js/src/features/human-resouce/components/Position/PositionForm.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { Form, Button, Input } from "antd";
+import { FormErrorMap } from "MyTypes";
+import { useFormServerErrors } from "hooks/useFormServerErrors";
+
+interface Props {
+  errors: FormErrorMap;
+  applyPosition: (text: string) => any;
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/Position/PositionForm.tsx:8]: void here, not any

File: js/src/features/human-resouce/components/Salary/SalaryForm.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { Form, Button, Input } from "antd";
+import { ApplySalaryForm } from "HumanResourceModels";
+import TextArea from "antd/lib/input/TextArea";
+import { FormErrorMap } from "MyTypes";
+import { useFormServerErrors } from "hooks/useFormServerErrors";
+
+interface Props {
+  applySalary: (data: ApplySalaryForm) => any;
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/Salary/SalaryForm.tsx:9]: void

File: js/src/features/human-resouce/components/Salary/SalaryForm.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  return (
+    <Form form={form} onFinish={handleFinish}>
+      <Form.Item
+        name="amount"
+        rules={[
+          { required: true, message: "Please input amount!" },
+          { pattern: /^\d+$/g, message: "Only digits are allowed!" },
+        ]}
+      >
+        <Input placeholder="Enter amount" />
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/Salary/SalaryForm.tsx:29]: If you only need numbers, then use InputNumber

File: js/src/features/human-resouce/components/Salary/SalaryForm.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      </Form.Item>
+      <Form.Item name="comment">
+        <TextArea placeholder="Enter comment" />
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/Salary/SalaryForm.tsx:32]: You can write Input.TextArea and you won’t need to import TextArea

File: js/src/features/human-resouce/epics.ts [hunk 8]
```diff
@@ -238,15 +241,13 @@ export const specifySkillsEpic: RootEpic = (action$, state$, { api }) =>
...
     filter(isActionOf(addEmployeeNoteAsync.request)),
-    debounceTime(500),
     switchMap((action) =>
       from(api.humanResource.addNote(action.payload)).pipe(
-        map((result) =>
+        mergeMap((result) =>
           result.success
-            ? addEmployeeNoteAsync.success()
-            : addEmployeeNoteAsync.failure(result.reason),
+            ? of(addEmployeeNoteAsync.success(), closeCreateNoteModal())
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/epics.ts:248]: If you need to return multiple actions, use [addEmployeeNoteAsync.success (), closeCreateNoteModal ()], because mergeMap returns an Observable

File: js/src/services/human-resource-api.ts [hunk 9]
```diff
@@ -96,7 +100,10 @@ export const applyPosition = ({
       success: true,
       data: null,
     }))
-    .catch<Failure>(() => ({ success: false, reason: "Error" }));
+    .catch<FormFailure>((resp) => ({
+      success: false,
+      errors: resp?.response?.data?.errors || [],
+    }));
 
 export const applySalary = ({ userId, ...body }: ApplySalaryRequestParams) =>
```
Comment (Aliaksandrzaz) [inline js/src/services/human-resource-api.ts:108]: Why use Spread for body?

File: js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx [hunk 10]
```diff
@@ -2,29 +2,29 @@ import React from "react";
...
 }) => (
   <Timeline mode="left">
     {employeeLocations.map((item) => (
       <Timeline.Item
         key={item.id}
-        label={`${item.startedAt} - ${item.endedAt || 'till now'}`}
-        color={item.endedAt ? "blue" : "green"}
+        label={`${item.startedAt} - ${item.endedAt || "till now"}`}
+        color={
+          isCurrentInterval(item.startedAt, item.endedAt) ? "green" : "blue"
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx:22]: Why use isCurrentInterval if the simple condition is item.endedAt? "blue": "green"

File: js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx [hunk 11]
```diff
@@ -2,29 +2,29 @@ import React from "react";
 import { Timeline } from "antd";
 import { EmployeeLocation } from "LocationModels";
 import { EmployeeLocationListInner } from "./EmployeeLocationListInner";
+import { isCurrentInterval } from "./helper";
 
 interface Props {
   employeeLocations: EmployeeLocation[];
   openEmployeeLocationEditForm: (val: EmployeeLocation) => void;
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx:9]: What does "val" mean?

File: js/src/features/human-resouce/components/Note/NoteForm.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        name="text"
+        rules={[
+          { required: true, message: "Please input message!" },
+          {
+            max: TEXT_LENGTH,
+            message: `Note text is too long. Max length is ${TEXT_LENGTH}`,
+          },
+        ]}
+      >
+        <TextArea rows={4} placeholder="Enter note text" />
```
Comment (Aliaksandrzaz) [inline js/src/features/human-resouce/components/Note/NoteForm.tsx:33]: You can write Input.TextArea and you won’t need to import TextArea

File: js/src/services/human-resource-api.ts (hunk 9 above)
Comment (DmitryKus) [reply js/src/services/human-resource-api.ts:108]: because all params are a single object. one property userId I need in the url and other props - amount and comment in body.

File: js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx (hunk 11 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx:9]: value

File: js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationListInner.tsx (hunk 2 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationListInner.tsx:7]: value

File: js/src/features/human-resouce/actions.ts (hunk 1 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/actions.ts:161]: I renamed it because before form was inline and now it's in modal. I think it's ok. I have in state  `isCreateModalVisible` and it is a corresponding action.

File: js/src/features/human-resouce/components/Note/NoteForm.tsx (hunk 3 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/Note/NoteForm.tsx:8]: done

File: js/src/features/human-resouce/components/Position/PositionForm.tsx (hunk 4 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/Position/PositionForm.tsx:8]: done

File: js/src/features/human-resouce/components/Salary/SalaryForm.tsx (hunk 5 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/Salary/SalaryForm.tsx:9]: done

File: js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx (hunk 10 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/EmployeeLocation/EmployeeLocationList.tsx:22]: because last location isn't current

File: js/src/features/human-resouce/components/Note/NoteForm.tsx (hunk 12 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/Note/NoteForm.tsx:33]: done

File: js/src/features/human-resouce/components/Salary/SalaryForm.tsx (hunk 6 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/Salary/SalaryForm.tsx:29]: done

File: js/src/features/human-resouce/components/Salary/SalaryForm.tsx (hunk 7 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/components/Salary/SalaryForm.tsx:32]: done

File: js/src/features/human-resouce/epics.ts (hunk 8 above)
Comment (DmitryKus) [reply js/src/features/human-resouce/epics.ts:248]: done
